### ISSUE TITLE
xgb.update bug with multiple models

### ISSUE THREAD

**Issue by vitaliyradchenko** [dmlc/xgboost#2564]
When I try to create different models and update them sequentially, they have different results on the same data.

In the example below, we create 20 identical models and update them. Then evaluate results and get different results. The first model gives a right result, but others seem to have been trained with another random seed.

`from sklearn.datasets import load_boston
import xgboost as xgb
param = {}
param['objective'] = 'reg:linear'
param['max_depth'] = 6
param['eta'] = .1
param['colsample_bytree'] = .5
param['subsample'] = .8
param['silent'] = 1
param['eval_metric'] = "mae"
df = load_boston()
xgb_train = xgb.DMatrix( df["data"][:-100,:], label=df["target"][:-100])
xgb_val = xgb.DMatrix( df["data"][-100:,:], label=df["target"][-100:])
models = []
for i in range(20):
    models.append(xgb.train(param, xgb_train, 1))

print(models[-1].eval(xgb_val))

for i in range(20):
    models[i].update(xgb_train, 2)
    print(models[i].eval(xgb_val))

model = xgb.train(param, xgb_train, 2)
print(model.eval(xgb_val))`

**Comment by khotilov** [comment]
Each model is initially configured with the default random seed=0. I.e., the state of the global random engine is reset for every model. However, for subsequent updates, the global random state is not reset after each update, and you get different models due to the requested sampling. Note that random state currently may only be seeded during model initialization (i.e., `models[i].set_param('seed',1)` before each update would have no effect).
